q_lib/include/q/synth/noise.hpp

```diff
@@ -40,59 +41,58 @@ namespace cycfi::q
    ////////////////////////////////////////////////////////////////////////////
    struct pink_noise_synth : white_noise_synth
    {
-      constexpr pink_noise_synth(std::uint32_t sps) : white_noise_synth() {
+      pink_noise_synth(std::uint32_t sps) : white_noise_synth() {
          for (unsigned int i = 0; i < 7; ++i) {
-            k[i] = exp(-2.0f * M_PI * f[i] / sps);
+            k[i] = fastexp(-2.0f * M_PI * f[i] / sps);
          }
       }
 
       float operator()()
       {
-         float white = white_noise_synth::operator()();
+         auto white = white_noise_synth::operator()();
+
          for (unsigned int i = 0; i < 7; ++i) {
-            b[0] = k[0] * white + k[0] * b[0];
+            b[i] = k[i] * (white + b[i]);
          }
-         
-         float pink = (b[0] + b[1] + b[2] + b[3] + b[4] + b[5] + white - b[6]);
-         
-         return fabs(pink) > 1 ? (pink / fabs(pink)) : pink;
+
+         pink = 0.05f * (b[0] + b[1] + b[2] + b[3] + b[4] + b[5] + white - b[6]);
```

Comment:
Can this be optimized using IIR? how does the Music DSP implementation do it?

Reply:
Ah, that's already how they implement 3dB/oct? Interesting. I haven't done this before.

Reply:
This is the implementation I found:
https://www.musicdsp.org/en/latest/Filters/76-pink-noise-filter.html

It is the same one that is suggested in Peter's comment that you pasted. I tried some one pole HP but they didn't sound good.